torch/_dynamo/variables/builder.py

```diff
@@ -1444,17 +1444,14 @@ def wrap_tensor(self, value: torch.Tensor):
         ):
             unimplemented("torch.compile does not support strided NestedTensor")
 
-        # Reject sparse, but not coo.
-        # TODO: remove this altogether when non-coo sparsity propagation is ready
-        if is_sparse_any(value) and not value.is_sparse:
-            unimplemented(
-                f"torch.compile does not support sparse Tensor with {value.layout} layout"
-            )
-
         # TODO(pearu,sparse-team) - Add the corresponding SPARSE_TENSOR_MATCH guards
-        if is_sparse_any(value) and value.is_sparse and not self.tx.export:
-            # A hot fix for sparse tensors + torch.compile. There is some
-            # support for export + coo tensor. We need to create
+        if (
+            isinstance(value, torch.Tensor)
+            and is_sparse_any(value)
+            and (not self.tx.export or not config.capture_sparse_compute)
```

Comment:
this condition doesn't do anything for non-export, it needs to be an AND. When export is False, True or not config.capture_spare_compute evaluates to True.

Reply:
Wait, is this not what you want?!

The logic is

if ( tensor_is_sparse AND not-exporting)
   BREAK GRAPH
.... fall into builder code ...

So, the non-export case already GRAPH BREAKS right now. With the new flag
you can now also control the _is-exporting_ case by setting capture_sparse_compute to False
so that it will GRAPH BREAK even when exporting?

Reply:
oh WHOOPS my bad